README.md

```diff
@@ -44,9 +50,110 @@ As the salt and hash are typically raw bytes, they also need to be converted
 to characters, for example using base64.
 
 All of the Passwap supplied algorithms use dollar sign (`$`) delimited
-encoding. This results in a single string containing all of the above for
+encoding, aka [Modular Crypt Format](https://passlib.readthedocs.io/en/stable/modular_crypt_format.htm).
+This results in a single string containing all of the above for
 later password verification.
 
+#### Argon2
+
+Argon2 uses standard raw base64 encoding (no padding) for salt and hash.
+The resulting Modular Crypt Format string looks as follows:
+
+```
+$argon2i$v=19$m=4096,t=3,p=1$cmFuZG9tc2FsdGlzaGFyZA$YMvo8AUoNtnKYGqeODruCjHdiEbl1pKL2MsYy9VgU/E
+   (1)              (2)               (3)                            (4)
+```
+
+Where:
+
+1. Identifier, which can be `argon2i` or `argon2id`. `argon2d` is not supported by Go and therefore not this library.
+2. Cost parameters.
+   1. `m` for memory, `4096` KiB in this example.
+   2. `t` for time, `3` in this example.
+   3. `p` for parallelism (threads), `1` in this example.
+3. Base64 encoded Salt.
+4. Base64 encoded Argon2 hash output of the password and salt combined.
+
+Changing any of the parameters or salt produces a different hash output.
+More information about the parameters can be found in the upstream [argon2 package documentation](https://pkg.go.dev/golang.org/x/crypto/argon2).
+
+### Bcrypt
+
+Bcrypt uses a custom base64 encoding with the character set of `[./A-Za-z0-9]` and padding.
+The actual formatting is fully implemented by the [Go package](https://pkg.go.dev/golang.org/x/crypto/bcrypt).
+The resulting Modular Crypt Format string looks as follows:
+
+```
+$2a$12$aLYFkieuqJyeynvptPTxpehSViui5WeAPuR2Xw1wui9CPHEaacmFq
+ (1)(2)          (3)                      (4)
+```
+
+Where:
+
+1. Identifier, which can be `2a`, `2b` or `2y`. It indicates the bcrypt version but is in fact ignored and the same is always produced.
+2. Cost parameter that is exponential. `12` in this example.
+3. Base64 encoded salt, always 22 character long.
+4. Base86 encoded Bcrypt hash output of the password and salt combined.
+
+
+### md5
+
+md5 uses its own encoding scheme part of the [hashing algorithm](https://passlib.readthedocs.io/en/stable/lib/passlib.hash.md5_crypt.html#algorithm). It uses a similar alphabet as Base64 but does additional shuffling of bytes.
+The resulting Modular Crypt Format string looks as follows:
+
+```
+$1$kJ4QkJaQ$3EbD/pJddrq5HW3mpZ4KZ1
+(1)   (2)           (3)
+```
+
+Where:
+
+1. Identifier, which is always `1`
+2. Base64-ish encoded salt.
+3. Base64-ish encoded md5 hash output of the password and salt combined.
+
+There is no cost parameter for md5. md5 is old, considered too light and insecure. It is provided to verify and migrate to a better algorithm. Do not use for new hashes.
```

Comment:
```suggestion
There is no cost parameter for MD5 because MD5 is old and is considered too light and insecure. It is provided to verify and migrate to a better algorithm. Do not use for new hashes.
```